examples/GraphTraversal/Graph.roc

```diff
@@ -0,0 +1,150 @@
+## The Graph interface represents a graph using an adjacency list and exposes
+## functions for working with graphs, such as creating one from a list and
+## performing a depth-first or breadth-first search.
+interface Graph
+    exposes [
+        Graph,
+        fromList,
+        dfs,
+        bfs,
+    ]
+    imports [
+        Dict,
+    ]
+
+## Graph type representing a graph as a dictionary of adjacency lists
+## where each key is a vertex and each value is a list of its adjacent vertices.
+Graph a := Dict a (List a) | a has Eq
+
+## Create a Graph from an adjacency list.
+fromList : List (a, List a) -> Graph a
+fromList = \adjacencyList ->
+    emptyDict = Dict.withCapacity (List.len adjacencyList)
+
+    update = \dict, (vertex, edges) ->
+        Dict.insert dict vertex edges
+
+    List.walk adjacencyList emptyDict update
+    |> @Graph
+
+## Perform a depth-first search on a graph to find a target vertex.
+##
+## - `isTarget` : A function that returns true if a vertex is the target.
+## - `root`     : The starting vertex for the search.
+## - `graph`    : The graph to perform the search on.
+dfs : (a -> Bool), a, Graph a -> Result a [NotFound]
+dfs = \isTarget, root, @Graph graph ->
+    dfsHelper isTarget [root] [] graph
+
+# A helper function for performing the depth-first search.
+#
+# `isTarget` : A function that returns true if a vertex is the target.
+# `stack`    : A list of vertices to visit.
+# `visited`  : A list of visited vertices.
+# `graph`    : The graph to perform the search on.
+dfsHelper : (a -> Bool), List a, List a, Dict a (List a) -> Result a [NotFound]
+dfsHelper = \isTarget, stack, visited, graph ->
+    when stack is
+        [] ->
+            Err NotFound
+
+        [current, ..] ->
+            rest = List.dropFirst stack
```

Comment:
Preferably use the other end of the list. It will be faster. Not sure if you can pattern match the last item of a list in roc. If not, you can just use `List.last` and then `List.dropLast`.

Reply:
Yep, fixed.